core/pkg/launch/config_files.go

```diff
@@ -0,0 +1,58 @@
+package launch
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/wandb/wandb/core/internal/data_types"
+	"github.com/wandb/wandb/core/pkg/service"
+)
+
+// Represents a config file parameter for a job.
+//
+// The relpath is the path to the config file relative to the files directory.
+// The includePaths and excludePaths are used to filter down the run config
+// before it is converted to a schema and saved as job inputs.
+type configFileParameter struct {
+	relpath      string
+	includePaths []ConfigPath
+	excludePaths []ConfigPath
+}
+
+func newFileInputFromRequest(
+	request *service.JobInputRequest,
+) (*configFileParameter, error) {
+	includePaths := make([]ConfigPath, len(request.GetIncludePaths()))
+	excludePaths := make([]ConfigPath, len(request.GetExcludePaths()))
+	for i, path := range request.GetIncludePaths() {
+		includePaths[i] = ConfigPath(path.Path)
+	}
+	for i, path := range request.GetExcludePaths() {
+		excludePaths[i] = ConfigPath(path.Path)
+	}
+	switch source := request.GetSource().GetSource().(type) {
+	case *service.JobInputSource_File:
+		return &configFileParameter{
+			source.File.GetPath(), includePaths, excludePaths}, nil
+	default:
+		return nil, fmt.Errorf("jobBuilder: invalid source type for file input")
```

Comment:
should the type check be the first thing you do?

Reply:
to allow for early return in case of an error

Reply:
This errror should never be happen but good point nonetheless, I've moved it up